core/src/main/scala/org/bitcoins/core/protocol/BtcHumanReadablePart.scala

```diff
@@ -0,0 +1,60 @@
+package org.bitcoins.core.protocol
+
+import org.bitcoins.core.config.{MainNet, NetworkParameters, RegTest, TestNet3}
+import org.bitcoins.core.util.Bech32HumanReadablePart
+
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Represents the HumanReadablePart of a Bech32 address
+  * [[https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki]]
+  */
+sealed abstract class BtcHumanReadablePart extends Bech32HumanReadablePart {
+  def network: Option[NetworkParameters]
```

Comment:
I think network should be a required field and we should disregard what Pieter Wuille said, it really doesn't make sense to have a hrp that isn't associated with a network IMO.